docs/specs/bot-protection.md

```diff
@@ -0,0 +1,448 @@
+- `bot_protection.fail_open`: If it is true, then if the challenge-based provider is service unavailable, access is granted. It is false by default.
+- `bot_protection.provider.type`: If `mode` is not `never`, then it is required. Specify the challenge-based provider to be used in this branch.
+- `bot_protection.risk_assessment.enabled`: Whether risk assessment is enabled.
+- `bot_protection.risk_assessment.provider.type`: It `enabled` is true, then it is required. Specify the risk assessment provider to be used in this branch.
+
+For example,
+
+```yaml
+authentication_flow:
+  login_flows:
+  - name: default
+    steps:
+    - type: identify
+      one_of:
+      - identification: email
+        # Identify with email requires bot protection.
+        bot_protection:
+          mode: "always"
+          provider:
+            type: cloudflare
+    - type: authenticate
+      one_of:
+      - authentication: primary_password
+      - authentication: primary_oob_otp_email
+```
+
+### Behavior of generated flows
+
+Given `bot_protection.risk_assessment.enabled=true`,
+
+1. All the branches of the first step (that is, the `identify` step, or the `authenticate` step in reauth flow) has `bot_protection.risk_assessment.enabled=true`.
+2. The configured provider is used as `bot_protection.risk_assessment.provider.type`
+
+The bot protection behavior in generated flows depend on
+
+- `bot_protection.enabled`
+- `bot_protection.authentication_flow`
+- `bot_protection.authenticator`
+
+Given `bot_protection.enabled=true`,
+
+|Configuration|Behavior|
+|---|---|
+|`bot_protection.authentication_flow.signup_or_login.mode`|Branches `identification: email`, `identification: phone`, `identification: username` of step `type: identify` in flows `signup`, `login`, `signup_login`, `promote` has `bot_protection.mode` equal to `bot_protection.authentication_flow.signup_or_login.mode`.|
+|`bot_protection.authentication_flow.account_recovery.mode`|All branches of `type: identify` of flow `account_recovery` has `bot_protection.mode` equal to `bot_protection.authentication_flow.account_recovery.mode`.|
+|`bot_protection.authenticator.password.mode`|Branches `authentication: primary_password`, `authentication: secondary_password` in step `type: authenticate` in flows `login` has `bot_protection.mode` equal to `bot_protection.authenticator.password.mode`.|
+|`bot_protection.authenticator.oob_otp_email.mode`|Branches `authentication: primary_oob_otp_email`, `authentication: secondary_oob_otp_email` in steps `type: authenticate`, `type: create_authenticator` in flows `signup`, `login`, `promote` has `bot_protection.mode` equal to `bot_protection.authenticator.oob_otp_email.mode`.|
+|`verification.claims.email.required`|If `verification.claims.email.required` is true, then branches `identification: email` in step `type: identify` in flows `signup`, `promote` has `bot_protection.mode` equal to `bot_protection.authenticator.oob_otp_email.mode`.|
```

Comment:
Seems this conflicts with `bot_protection.authentication_flow.signup_or_login.mode`

What should

> branch `identification: email` in step `type: identify` in flows `signup`, `promote` 

be, in case of below config?

```
bot_protection.authentication_flow.signup_or_login.mode: never
bot_protection.authenticator.oob_otp_email.mode: always
verification.claims.email.required: true
```

Maybe add below?

```md
#### Behavior of generated flow in case of conflicts in configuration
In case of any conflicts, the stricter [Risk Mode](#risk-level-mode) will be taken, where 

     `always` > `risk_level_high` > `risk_level_medium` > `never` 

in terms of strictness
```

Reply:
Yes, you're correct.

Reply:
Added commit "eClarify what mode is used when there are multiple contributions"